hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/logging/UnknownMethodAlgorithmLogger.java

```diff
@@ -37,98 +37,73 @@
+                    MessageDigestAlgorithm messageDigestAlgorithm = signedInfo.item(0).getMessageDigestAlgorithm();
+                    if (messageDigestAlgorithm != null) {
+                        return messageDigestAlgorithm.getJCEAlgorithmString();
                     }
-                } catch (XMLSecurityException e) {
-                    LOG.debug(format("Error getting message digest algorithm: {0}", e));
                 }
+            } catch (XMLSecurityException e) {
+                LOG.debug("Error getting message digest algorithm", e);
             }
         }
         return null;
     }
 
     private static void logMethodAlgorithm(
             final Role role,
-            final String signatureMethodAlgorithm,
-            final String digestMethodAlgorithm,
+            final Signature signature,
             final String location) {
+        final String signatureMethodAlgorithm = signature.getSignatureAlgorithm();
+        final String digestMethodAlgorithm = getDigestMethodAlgorithm(signature);
         final boolean validSignatureAlgorithm = VALID_SIGNATURE_ALGORITHMS.contains(signatureMethodAlgorithm);
         final boolean validDigestAlgorithm = VALID_DIGEST_ALGORITHMS.contains(digestMethodAlgorithm);
         if (!validSignatureAlgorithm && !validDigestAlgorithm) {
-            LOG.info(
-                String.format(SIGNATURE_AND_DIGEST_ALGORITHMS_MESSAGE, role, signatureMethodAlgorithm, digestMethodAlgorithm, location)
-            );
-        }
-        else if (!validSignatureAlgorithm) {
-            LOG.info(
-                    String.format(SIGNATURE_ALGORITHM_MESSAGE, role, signatureMethodAlgorithm, location)
-            );
-        }
-        else if (!validDigestAlgorithm) {
-            LOG.info(
-                    String.format(DIGEST_ALGORITHM_MESSAGE, role, digestMethodAlgorithm, location)
-            );
+            LOG.info(format(SIGNATURE_AND_DIGEST_ALGORITHMS_MESSAGE, role, signatureMethodAlgorithm, digestMethodAlgorithm, location));
+        } else if (!validSignatureAlgorithm) {
+            LOG.info(format(SIGNATURE_ALGORITHM_MESSAGE, role, signatureMethodAlgorithm, location));
+        } else if (!validDigestAlgorithm) {
+            LOG.info(format(DIGEST_ALGORITHM_MESSAGE, role, digestMethodAlgorithm, location));
         }
     }
 
     public static void probeResponseForMethodAlgorithm(final InboundResponseFromIdp response) {
         if (response != null) {
-            final Optional<Signature> signature = response.getSignature();
-            if (signature != null) {
-                final String signatureMethodAlgorithm = getSignatureMethodAlgorithm(signature);
-                final String digestMethodAlgorithm = getDigestMethodAlgorithm(signature);
-                logMethodAlgorithm(Role.IDP, signatureMethodAlgorithm, digestMethodAlgorithm, Response.DEFAULT_ELEMENT_LOCAL_NAME);
-            }
+            response.getSignature().ifPresent((signature) ->
```

Comment:
Maybe this could throw an NPE now actually

Reply:
Think the `response` signature is validated prior to this call so `response.getSignature()` should be null safe